### DISCUSSION TITLE
Debrief and record data on clic

### DISCUSSION THREAD

**Discussion by Julie-DW** [jspsych/jsPsych#1043] [Q&A] [answered]
Hello everyone, 

I have created a JsPsych experiment for my psychology Phd studies. I recently put this experiment on cognition.run, and, at the end of this experiment, I would like to show a debrief text to participants (thank you for your participation for example) and then show a link to join another platform to complete a questionnaire online (LimeSurvey). For that, I put a "html-keyboard-response" var debrief. 

The problem I met, is when the particpant clic on the link to complete the questionnaire, the experiment appears not finished on cognition.run and the results downoalded on cognition.run could be uncomplete. 

I would like to avoid this by knowing if it is possible to consider the task is done when the participants clic on the link ? And if it is possible, where and what should I add to the JsPsych code (which is as below for the moment) ? 

```js
[...]

jsPsych.init({
     timeline: [survey_page, pre_trial1, pre_trial2, ...timeline1, pre_trial3, ...timeline2, pre_trial4, debrief],
     on_finish: function() { //Execute this when the experiment finishes
       jsPsych.data.get().localSave('csv', 'testSave.csv'); //Save the data locally in a .csv file
       jsPsych.data.displayData(); //Display the data onto the browser screen
     }
   });
```

In advance, thank you very much for your help, 
Bests, 
Julie

**Comment by yuvalharr** [comment, marked as answer]
Hey there Julie,

I think that what you are looking for is something like this:
```
var hello_trial = {
    type: 'html-keyboard-response',
    stimulus: 'Hello world!'
}

jsPsych.init({
    timeline: [hello_trial],
    on_finish: function() {
      window.location.href = "https://YourLinkHere.com/"
    }
})
```
The on_finish function will redirect to whichever site you'd like, while still finishing the current experiment.

Also, I'm not sure if there is any need for you to use the jsPsych.data.get().localSave('csv', 'testSave.csv') function, as Cognition.run should save all of the data for you even without it.

Cheers, 
Yuval

**Comment by Julie-DW** [comment]
Hi Yuval, 

Thank you very much for taking time to answer me, indeed it works now perfectly and you are right about the localSave function. 
Thank you very much for your help !! 
Have a good day, 
Bests, 

Julie
